client/boot/index.js

```diff
@@ -183,16 +184,31 @@ function reduxStoreReady( reduxStore ) {
 			translatorInvitation: translatorInvitation
 		} );
 	} else {
+		let props = {};
 		analytics.setSuperProps( superProps );
 
-		if ( startsWith( window.location.pathname, '/design' ) ) {
+		// TODO(ehg): Delete this mini-router when we have an isomorphic, single render tree routing solution
+		if ( config.isEnabled( 'manage/themes/details' ) ) {
+			const themesRoutes = [
+				{ name: 'design', path: new Path( '/design' ) },
+				{ name: 'themes', path: new Path( '/themes/:theme_slug' ) },
+			];
+
+			const matchedRoutes = themesRoutes
+				.map( r => ( Object.assign( {}, r, { match: r.path.partialMatch( window.location.pathname ) } ) ) )
+				.filter( r => r.match !== null );
+
+			if ( matchedRoutes.length ) {
+				props = { routeName: matchedRoutes[0].name, match: matchedRoutes[0].match };
+				Layout = require( 'layout/logged-out-design' );
+			}
+		} else if ( startsWith( '/design', window.location.pathname ) ) {
```

Comment:
Isn't this the wrong arg order? cf [L188](https://github.com/Automattic/wp-calypso/pull/3053/files#diff-1434133b7ef90d16331d0125afb1fadcL188) and https://lodash.com/docs#startsWith